### PR TITLE
Set the correct cause of the verification failure

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We should support setting the exact cause of the verification failure which may help the remote peer to fix it.

Modifications:

Change CertificateVerifier to allow set the actual cause.

Result:

Give more control about certificate verification

### CODE REVIEW

**Comment by normanmaurer** [comment]
@nmittler @Scottmitch PTAL

**Comment by nmittler** [comment]
LGTM

**Comment by normanmaurer** [comment]
Cherry-picked into 1.1.33 as (26fa1429c92238ef28e2e6b2d93112bfa1063ef4)

File: openssl-dynamic/src/main/java/org/apache/tomcat/jni/CertificateVerifier.java [hunk 1]
```diff
@@ -20,6 +20,73 @@
  */
 public interface CertificateVerifier {
 
+    int X509_V_OK = 0;
```
Comment (Scottmitch) [inline openssl-dynamic/src/main/java/org/apache/tomcat/jni/CertificateVerifier.java]: would be nice to have a link as to where these were derived from ... I suspect some openssl header file though :wink:

**Comment by Scottmitch** [comment]
lgtm

File: openssl-dynamic/src/main/java/org/apache/tomcat/jni/CertificateVerifier.java (hunk 1 above)
Comment (normanmaurer) [inline openssl-dynamic/src/main/java/org/apache/tomcat/jni/CertificateVerifier.java]: haha yeah... ;) Let me add a link

Comment (normanmaurer) [inline openssl-dynamic/src/main/java/org/apache/tomcat/jni/CertificateVerifier.java]: @Scottmitch actually these are from: 
https://www.openssl.org/docs/manmaster/apps/verify.html

Comment (Scottmitch) [inline openssl-dynamic/src/main/java/org/apache/tomcat/jni/CertificateVerifier.java]: OooO docs ... i guess they would also live in some SSL / TLS rfc somewhere too.

Comment (normanmaurer) [inline openssl-dynamic/src/main/java/org/apache/tomcat/jni/CertificateVerifier.java]: :+1: I will check and add a follow pr

**Pull request closed without merging**